common/src/main/java/io/pravega/common/concurrent/FutureHelpers.java

```diff
@@ -252,6 +252,28 @@
     }
 
     /**
+     * Attempts to retrieve the failure cause of a Future. This is equivalent to calling get() on the Future and catching
+     * and unwrapping any exceptions being thrown.
+     *
+     * @param future The Future to check.
+     * @return A Throwable indicating the failure cause, or null if the Future completed normally.
+     * @throws IllegalArgumentException If the Future is not yet completed.
+     */
+    public static Throwable getFailureCause(CompletableFuture<?> future) {
```

Comment:
I currently have a PR which adds a very similar method:
https://github.com/pravega/pravega/pull/1764/files#diff-d89cb9f56cd6e22054026ff28b722371R120

We should reconcile these down to one version. They are semantically different in what they do if the future has completed successfully. I think this should be treated the same as though it was still running: If there is no exception, it should return null. If you move you're method up to just below isSuccessful on line 120 above, it will force a merge conflict for us to resolve reguardless of which PR goes in first.